### ISSUE TITLE
Include ffmpeg

### ISSUE THREAD

**Issue by peterkronenberg** [openpnp/opencv#74]
According to https://forum.opencv.org/t/error-in-java-cant-find-starting-number-in-the-name-of-file/3014/2, I need opencv_videoio_ffmpeg452_64.dll.  Is there any reason it's not part of the packaging?

**Comment by vonnieda** [comment]
Please see other issues in this repo with regards to ffmpeg. In brief: I don't know of a way to statically link it into the binary, and it is a lot of work to try to include it. I'm open to PRs to the Github Actions scripts that fix this.

**Comment by peterkronenberg** [comment]
Thanks.  I just saw the other issue.  Would be nice if you at least documented this and explained exactly how to set things up and where to get the correct version of ffmpeg.   And are these other cavets as well, where the maven version is not complete?

I'm still having a problem and can't get it to work, even if I tried to point to ffmpeg using -Djava.library.path or -Djna.library.path.  According to the issue I opened on opencv, these won't work, because Java does not load the file.  I need to put the file on the class path, but can't get that working either.

**Comment by peterkronenberg** [comment]
Figured out my problem.  I thought the dll needed to be on the classpath, but it needs to be on the path (or in the same directory as the application)

**Comment by lynnwilliam** [comment]
> Figured out my problem. I thought the dll needed to be on the classpath, but it needs to be on the path (or in the same directory as the application)

Can you give a specific example of this please? 
I got the dll required and want to know where you placed them relative to the other dlls in the project ?
Could you paste the paths please?

**Comment by gevant** [comment]
@vonnieda  Just did some testing with interesting results: 
in OpenCV 4.60 and 4.70 something changed on a way how opencv_videoio_ffmpeg4xx_64.dll is compiled and/or linked to opencv_java4xx.dll. 
I tested with 4.2.0- 4.5.0 all versions do work with a ip camera, your solution to extract binaries out of a jarfile into the temp folder and set  java.library.path internally would work when corresponding ffmpeg4 dll would be included, with not much additional effort. BUT 2 latest open cv distributions are compiled differently and it won't work as distributed.
My tests were performed on windows 10pro

**Comment by lynnwilliam** [comment]
I got webcam and RTSP working this morning.
Confirmed in Windows 10 and windows 11.

**Step 1**
//this did NOT work using a TEMP File.
// I have other code that takes the opencv_videoio_ffmpeg460_64.dll file from my /resources directly and copies it over to the users /documents directory.
So do this first if you need, I did not add the code from copying the file over, since its boiler plate stuff.

**Step 2**
use the windows driver
opencv_videoio_ffmpeg460_64.dll
from the OpenCV.exe distribution.
Than load that opencv_videoio_ffmpeg460_64.dll file from a folder with system access e.g "c:/users/.../documents/opencv_videoio_ffmpeg460_64.dll"

**Step 3**
using the code
Path path = //  the path to the absolute folder of the opencv_videoio_ffmpeg460_64.dll files
final String normalizedPath = path.normalize().toString();
final Field field = ClassLoader.class.getDeclaredField("usr_paths");
        field.setAccessible(true);
 final Set<String> userPaths = new HashSet<>(Arrays.asList((String[]) field.get(null)));
        userPaths.add(normalizedPath);
        field.set(null, userPaths.toArray(new String[userPaths.size()]));
        System.setProperty("java.library.path", System.getProperty("java.library.path") + File.pathSeparator + normalizedPath);

**Comment by vonnieda** [comment]
I'm consolidating other requests to include opencv-ffmpeg here, and will point others here in the future. If this is something you'd like to see implemented, please consider submitting or sponsoring a PR to implement it.

**Comment by lynnwilliam** [comment]
Yes you can just point others here and they can use this solution for Windows if they need to.
This code does work, and it is just to show that you can get windows DLLS working with your OpenCV project , without any modifications to the repo.
I think adding FFMPEG to OpenCV is a bigger request than it seems ( since you now have to support FFMPEG testing etc)
, I might open a pull request for this feature in the future.

**Comment by vonnieda** [comment]
> I think adding FFMPEG to OpenCV is a bigger request than it seems ( since you now have to support FFMPEG testing etc) , I might open a pull request for this feature in the future.

I agree. I have avoided it for years because of this. Maintaining this project is already a huge task: Building binaries for so many platforms is a ton of work. Adding more work just isn't something I'm super interested in, but PRs are *always* welcome!

**Comment by violet-isle** [comment]
Have you looked into how JavaCV does it? This issue is causing a project I am using to break on my Arch install.
